airflow/providers/google/cloud/utils/credentials_provider.py

```diff
@@ -419,3 +452,18 @@ def _get_project_id_from_service_account_email(service_account_email: str) -> st
         raise AirflowException(
             f"Could not extract project_id from service account's email: {service_account_email}."
         )
+
+
+def _get_info_from_credential_configuration_file(credential_configuration_file):
+    if isinstance(credential_configuration_file, str) and os.path.exists(credential_configuration_file):
+        with open(credential_configuration_file) as file_obj:
+            try:
+                info = json.load(file_obj)
+            except ValueError:
+                raise AirflowException("Credentials Configuration File is not a valid json file.")
+    else:
+        try:
+            info = json.loads(credential_configuration_file)
+        except json.decoder.JSONDecodeError:
+            raise AirflowException("Invalid JSON.")
```

Comment:
If I understand it correctly, the `credential_configuration_file` can be `str` or  `dict[str, str]`. How would it work in the case of a dictionary?
And this part is also missing unit test.

Reply:
Thank you for the heads up, the function logic seems to be incorrect. I have fixed it and we will soon be adding unit tests as well. Let me know if it looks better now